Framework/Kernel/src/TimeSeriesProperty.cpp

```diff
@@ -176,6 +176,9 @@ operator==(const TimeSeriesProperty<TYPE> &right) const {
   { // so vectors can go out of scope
     std::vector<DateAndTime> lhsTimes = this->timesAsVector();
     std::vector<DateAndTime> rhsTimes = right.timesAsVector();
+    if (lhsTimes.size() != rhsTimes.size()) {
```

Comment:
Shouldn't you use  `TimeSeriesProperty::size()` or `TimeSeriesProperty::realSize()` and copy the vectors if they aren't equal?